src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs

```diff
@@ -261,49 +306,91 @@ private string GetBuildConfiguration()
             buildProperties[DotNetCoreManifestFilePropertyKeys.StartupDllFileName] = startupDllFileName;
         }
 
-        public string GetInstallerScriptSnippet(BuildScriptGeneratorContext context)
+        public string ResolveVersion(string versionToResolve)
+        {
+            var resolvedVersion = GetVersionUsingHierarchicalRules(versionToResolve);
+            resolvedVersion = GetMaxSatisfyingVersionAndVerify(resolvedVersion);
+            return resolvedVersion;
+        }
```

Comment:
I made an IPlatformVersionResolver interface for keeping version resolving logics like this, and then use it from each Platform, we could discuss more about it!